### PR TITLE
Modify Torch Support to handle tensors on GPU

### PR DESCRIPTION
**Pull request by afspies**
It seems the torch support assumed tensors exist on CPU, as this is a pre-requisite for conversion to numpy arrays with `some_tensor.numpy()`. Perhaps this was an intentional design choice to avoid occupying non-accelerator memory for users who aren't being careful, but moving non-cpu tensors onto cpu automatically for the purposes of visualization is likely acceptable in most use cases, so I have added this.

Minor note that the .cpu() and .detach() conversion can be combined into one-line. Not sure what is preferable.

### CODE REVIEW

**Review by danieldjohnson — CHANGES_REQUESTED** [review]
Thanks for the PR! This was an oversight, I must not have thoroughly tested this with GPU tensors.

That said, I think it's better to avoid copying the entire array to CPU memory at the beginning, because the array may be very large and we may only need to visualize a small portion of it. Would you mind changing this to only copy to CPU right before converting to numpy? (Or, I think `.numpy(force=True)` accomplishes the same thing?)

**Comment by afspies** [comment]
Hi @danieldjohnson,

Thanks for your response - Indeed I only just noticed the slicing! Looking into the `force=true` flag indeed confirms this as a better option as it also handles some other cases (I didn't know that flag existed!)

Made the changes - not sure if this is needed in `get_array_data_with_truncation`, but it probably doesn't hurt as torch claims not to return copies unless some sophisticated conversion is performed.

**Comment by danieldjohnson** [comment]
Thanks!

One more minor thing: would you mind squashing these commits together into a single commit so that the old version doesn't appear in the commit history?

**Comment by afspies** [comment]
Done :)

**Review by danieldjohnson — APPROVED** [review]

File: treescope/external/torch_support.py [hunk 1]
```diff
@@ -74,12 +74,13 @@ def _truncate_and_copy(
       ignoring any axes whose slices are already computed in `source_slices`.
   """
   assert torch is not None, "PyTorch is not available."
+
```
Comment (danieldjohnson) [inline treescope/external/torch_support.py:77]: Actually, sorry, could you also amend to remove this line with extra whitespace? It's tripping up the internal lint system and I haven't set up the external lint checks yet.

**Review by danieldjohnson — CHANGES_REQUESTED** [review]

File: treescope/external/torch_support.py (hunk 1 above)
Comment (afspies) [reply treescope/external/torch_support.py]: Hopefully ok now!

Comment (danieldjohnson) [reply treescope/external/torch_support.py]: Thanks!

**Review by danieldjohnson — APPROVED** [review]
